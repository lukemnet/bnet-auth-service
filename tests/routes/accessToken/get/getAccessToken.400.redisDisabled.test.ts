import fastify from 'fastify';
import fastifyRedis from 'fastify-redis-mock';
import server from '../../../../src/index';

const config = {
  app: {
    nodeEnv: 'test',
    port: '8123',
  },
  bnet: {
    region: 'us',
    apiKey: 'key',
    apiSecret: 'secret',
  },
  redis: {
    enable: false,
    cacheSegment: 'test',
    replyCachePeriod: 100,
    host: '127.0.0.1',
    port: '6379',
    db: '0',
    password: '',
    enableReadyCheck: true,
    dropBufferSupport: false,
  }
}

const { BlizzAPI } = require('blizzapi');
BlizzAPI.prototype.getAccessToken = () => '';

describe('/accessToken/get 400 (Redis disabled)', () => {
  const fastifyServer = fastify();

  beforeAll(() => {
    fastifyServer.register(fastifyRedis, {
      host: '127.0.0.1',
      port: '6379',
      password: '',
      enableReadyCheck: true,
      dropBufferSupport: false,
    });
    fastifyServer.register(server, config);
  });

  afterAll(() => fastifyServer.close());

  it('returns 400', async () => {
    const res = await fastifyServer.inject({ method: 'GET', url: '/accessToken/get', });
    expect(res.statusCode).toBe(400);
  });

  it('returns correct response', async () => {
    const res = await fastifyServer.inject({ method: 'GET', url: '/accessToken/get', });
    expect(JSON.parse(res.payload)).toEqual({"status":400});
  });
});