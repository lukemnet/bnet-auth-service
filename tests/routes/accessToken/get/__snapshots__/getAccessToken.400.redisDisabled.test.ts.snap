// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/accessToken/get 400 (Redis disabled) returns correct response 1`] = `"{\\"status\\":400}"`;
